hpx/parallel/util/loop.hpp

```diff
@@ -96,23 +281,98 @@ namespace hpx { namespace parallel { namespace util
                 return it;
             }
         };
+
+        template <typename T>
+        struct loop_n <prefetching_iterator<T>>
+        {
+            /////////////////////////////////////////////////////////////////
+            // handle sequences of non-futures when using prefetching
+            template <typename F>
+            static prefetching_iterator<T> call(prefetching_iterator<T> it,
+                std::size_t count, F && f)
+            {
+                std::size_t cnt = count + 1;
+                for (/**/; cnt != 0; (void) --cnt, ++it)
+                {
+                    std::vector<std::size_t>::iterator inner_it = it.base;
+                    std::size_t j = it.idx;
+                    std::size_t last = it.idx+it.chunk_size;
+
+                    //based on HPX Inspection Report,
+                    //using std::min violates boost min/max
+                    if (it.range_size < last)
+                        last = it.range_size;
+
+                    for(/**/; j < last ; ++j)
+                    {
+                        f(inner_it);
+                        ++inner_it;
+                    }
+                    if(j < it.range_size - 1)
```

Comment:
Am 03.06.2016 5:17 nachm. schrieb "Hartmut Kaiser" <notifications@github.com
>:
>
> In hpx/parallel/util/loop.hpp:
>
> > +                {
> > +                    std::vector<std::size_t>::
> iterator inner_it = it.base;
> > +                    std::size_t j = it.idx;
> > +                    std::size_t last = it.idx+it.chunk_size;
> > +
> > +                    //based on HPX Inspection Report,
> > +                    //using std::min violates boost min/max
> > +                    if (it.range_size < last)
> > +                        last = it.range_size;
> > +
> > +                    for(/**/; j < last ; ++j)
> > +                    {
> > +                        f(inner_it);
> > +                        ++inner_it;
> > +                    }
> > +                    if(j < it.range_size - 1)
>
> I'd suggest to use prefetching intrinsics here, at least on those
platforms where they are available.

Absolutely. But they have to be properly guarded.

>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.